### ISSUE TITLE
Wrong XYZ to LAB conversions.

### ISSUE THREAD

**Issue by hgm314** [scikit-image/scikit-image#3888]
When trying to convert XYZ color values into LAB values I found wrong results.
I assume that your reference color conversion calculator is correct and that was one of the resources I used to verify the data. (http://www.easyrgb.com/en/convert.php#inputFORM)
I will just show an example below.

## Way to reproduce
import numpy as np
from skimage.color import xyz2lab

print xyz2lab(np.asarray([69.548, 72.297, 60.442]).reshape((1, 1, 3)), illuminant='D50')[0, 0]
[467.2420834   -1.61113858  -3.63089562]

#### Warning: a L* value > 100.0 does not make sense!

The aforementioned web page gives this result:
Select data type...
XYZ
D50
2º (1931)
Insert color data...
69.548
72.297
60.442
Convert...
XYZ (D50/2°) = 69.548 72.297 60.442

sRGB   0-255 =  219.776  221.239  222.454    D65/2°
sRGB   0-1.0 =  0.86187  0.86760  0.87237
RGB Adobe 98 =  219.218  220.285  221.477
HSL    0-1.0 =  0.57562  0.03951  0.86712   207.22°
HSV    0-1.0 =  0.57562  0.01204  0.87237
HSI    0-1.0 =  0.57562  0.00624  0.86728
CMY    0-1.0 =  0.13813  0.13240  0.12763
CMYK   0-1.0 =  0.01204  0.00546  0.00000  0.12763
CMYK   0-100 =    1.204    0.546    0.000   12.763
XYZ          =   69.548   72.297   60.442    D50/2°
Yxy          =   72.297  0.34381  0.35740
**CIE-L*ab     =   88.111   -0.348   -0.781**
CIE-L*Ch(ab) =   88.111    0.855  245.953°
CIE-L*uv     =   88.111   -0.948   -0.914
CIE-L*Ch(uv) =   88.111    1.317  223.946°
HunterLab    =   85.028   -0.343   -0.653

## Version information
```python
# Paste the output of the following python commands
from __future__ import print_function
import sys; print(sys.version)
import platform; print(platform.platform())
import skimage; print("scikit-image version: {}".format(skimage.__version__))
import numpy; print("numpy version: {}".format(numpy.__version__))
```

```python
# your output here
2.7.15rc1 (default, Nov 12 2018, 14:31:15) 
[GCC 7.3.0]
Linux-4.15.0-47-generic-x86_64-with-Ubuntu-18.04-bionic
scikit-image version: 0.14.2
numpy version: 1.16.0
```

**Comment by johanjeppsson** [comment]
I think that this is related to the expected range of the datatypes. In the case of floats, I think that scikit-image expects the range to be [0,1] (or [-1,1]), otherwise it uses the max value of the current dtype.

If I use the values for your example, I get the expected value if I convert your XYZ value (which I assume is in the range to [0, 100] to floats on range [0, 1] or uint8 on range [0, 255].

```python
import numpy as np
from skimage import color

XYZ = np.dstack([69.548, 72.297, 60.442])

print('L: {:.3f} a: {:.3f} b: {:.3f}'.format(*color.xyz2lab(XYZ, illuminant='D50').flatten()))
print('L: {:.3f} a: {:.3f} b: {:.3f}'.format(*color.xyz2lab(XYZ / 100, illuminant='D50').flatten()))
print('L: {:.3f} a: {:.3f} b: {:.3f}'.format(*color.xyz2lab((255 * XYZ / 100).astype(np.uint8), illuminant='D50').flatten()))
```
```
L: 467.242 a: -1.611 b: -3.631
L: 88.111 a: -0.347 b: -0.782
L: 88.044 a: -0.350 b: -0.849
```

**Comment by jni** [comment]
@johanjeppsson good catch, and thank you for chiming in! Given that it was in XYZ space I had thought that dtype would no longer matter.

I'm cross-referencing #3009 as this is another source of confusion that we should do better at avoiding.
